EtherealEngine/etherealengine
# Create /chat page with existing social features

## Summary

**FEATURES**

This PR includes the following:

- New /chat route using tailwind
- New FeathersHooks utility for Reactive feathers requests and caching
- replaces parties and groups with channels
- media instance connection for world and channels unified

**DATABASE MIGRATIONS**

- `party`, `party_user`, `group`, `group_user`, `group_user_rank` `channel_type` tables have been dropped
- `group:read` `group:write` `party:read` `party:write` scopes have been replaced with `channel:read` and `channel:write`
- `channel_user` table has been added
- `group` and `party` invite types are now `channel`
- `group` and `party` invite email subject fields collapsed into `channel`


## References

closes #_insert number here_


## Checklist
- [ ] [ensure all checks pass](https://github.com/etherealengine/etherealengine/wiki/Testing-&-Contributing)
- [ ] Changes reviewed by at least 2 approved reviewer


## QA Steps

- ensure admin panel channels page works correctly, populating data and search query 
- ensure socials work, messaging friends and starting and ending media sessions with them
